backend/src/resolvers/userResolvers.js

```diff
@@ -21,25 +21,19 @@ export default {
     },
   },
   Mutation: {
-    removeUser: async (_, { user }) => {
-      return await User.findByIdAndDelete(user);
-    },
-    updateUser: async (_, { id, userData }) => {
-      return await User.findByIdAndUpdate(id, userData);
-    },
-
-    signup: async (_, { userInput }) => {
-      const user = await User.register(
+    signup: async (_, { user }) => {
+      const u = await User.register(
         new User({
-          username: userInput.username,
-          email: userInput.email,
-          firstName: userInput.firstName,
-          lastName: userInput.lastName,
+          username: user.username,
+          email: user.email,
+          firstName: user.firstName,
+          lastName: user.lastName,
         }),
-        userInput.password
+        u.password
```

Comment:
This specific line has to be `user.password`. I'm fixing it for you. `u` isn't defined yet when you need that value.